### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'ImportFolderID')

### ISSUE THREAD

**Issue by whoiskevinrich** [ShokoAnime/Shoko-WebUI#631]
The UI will crash when opening the `/webui/utilities/unrecognized` route if files have been moved or removed. I can reliably reproduce the error by:
1. Add unrecognized file to import library
2. Run import
3. Remove unrecognized file from import library folder
4. Navigate to Unrecognized route

I have been able to reliably fix the error by running the "Remove Missing Files" action, and retrying.

I am running the docker server as directed from the documentation, the only custom configuration I'm using is a different port mapping for the host.

Server Version: 4.2.2.0
Server Channel: Stable
WebUI Version: 2.0.0-dev.67
WebUI Channel: Stable

Stack Trace:
```
Trace:

    at Components_UtilitiesTable ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2598350)
    at div
    at open_closed_C ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2067548)
    at {REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2073180
    at {REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2076193
    at components_TransitionDiv ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2077718)
    at div
    at ScrollSync ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1500195)
    at div
    at div
    at open_closed_C ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2067548)
    at {REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2073180
    at {REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2076193
    at components_TransitionDiv ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2077718)
    at UnrecognizedUtilityTabs_UnrecognizedTab ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2608470)
    at RenderedRoute ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1917436)
    at Outlet ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1920494)
    at div
    at div
    at Panels_ShokoPanel ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2205487)
    at utilities_UnrecognizedUtility ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2540037)
    at RenderedRoute ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1917436)
    at Outlet ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1920494)
    at div
    at utilities_UtilitiesPage
    at RenderedRoute ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1917436)
    at Outlet ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1920494)
    at div
    at div
    at div
    at main_MainPage ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2195984)
    at router_AuthenticatedRoute ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2203383)
    at RenderedRoute ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1917436)
    at RenderedRoute ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1917436)
    at Routes ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1921494)
    at dist_Router ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1920588)
    at ReduxRouter ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1925693)
    at div
    at router ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2890507)
    at errorboundary_ErrorBoundary ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2105614)
    at ErrorBoundary ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:2108987)
    at components_Provider ({REDACTED}/webui/dist/main.036f5b9b6b605d106577.js:2:1435645)
    at app
```

**Comment by revam** [comment]
Tl;dr: Downgrade from the old beta web UI to the latest atable veraion (1.1.12).

## The longer answer 

You are using an old and outdated **beta** version of the web UI that was accidentally shipped with the 4.2.2 release. This issue is fixed in newer **beta** versions and is not applicable to older **stable** versions of the web UI, but nonetheless thank you for taking the time to report this issue. You have three options for fixing this; a) ignore the issue and keep your current version of the outdated **beta** web UI, with all it's bugs and features, b) downgrade to the latest **stable** version of the web UI, losing a lot of the bugs and features, or c) switch your server and web UI over to the dev./daily channel and update to the latest daily server and **beta** web UI, giving you access to new and improved **bleeding edge** features and much faster bug fixes in exchange for a piece of your soul and locking you into the dev./daily channel until the next stable server update unless you restore your database from a backup¹. I've added instructions to downgrade your web UI using swagger² you can use us you choose to downgrade.

## How to revert the Web UI back to a working __stable__ version 101

- Open Swagger by navigating to `http://<shoko host>:8111/swagger/index.html?urls.primaryName=3.0`.  Be sure to replace `<shoko host>` with the IP or DNS name of your Shoko entry.

- Generate an API key to use by executing `POST /api/auth`, found under the "**Auth**" section in Swagger.
  Press the "Try it out" button, enter your username for `user`, your password for `pass`, and `"swagger"` for `device`, and press "**Execute**". Copy the API key (the value of the `"apikey"` field from the response).

- Assign the API key to the `ApiKey` field under the "**Authorize**" menu found at the top of the page.

- Execute `GET /api/v{version}/WebUI/Update`, found under the "**WebUI**" section in Swagger, making sure to select `Stable` for the `channel` and input `3` for the `version`.

- Reset the Web UI. Close the current tab, and open a new tab. Do not refresh the page. If you enabled "Remember Me" then you also have to clear the local storage in your browser for the site.

### How to clear local storage 102

- Open the **DevTools** in your browser. (You can do this by pressing **F12** or by going into the browser menu at the far right on the browser toolbar, then entering the "**More Tools**" sub-menu, and pressing "**Developer tools**" in **Chromium-based browsers** or "**Web Developer Tools**" in **Firefox-based browsers**.)

- Navigate to the "**Application**" tab in **Chromium-based browsers**, and "**Storage**" in **Firefox-based browsers**.

- Find "**Local Storage**" in the side-bar to the left and click it to reveal the available sites.

- Find the site matching the current site you're on (e.g. `http://localhost:8111`, etc.), and click it.

- **Chromium-based browsers**: In the panel to the right, click the ":no_entry_sign:" button between the "Filter" box and the ":x:" button. **Firefox-based browsers**: Right-click on any of the items in the right panel, then press "**Delete All**". If there are no entries on the page then you don't need to do anything.

And you're done, you can now close the tab and open a new tab and it should work as expected.

¹ _Shoko creates an automagic backup of your database everytime it tries to modify the database structure when starting the server._
² _Which you can also find these instructions pinned in our support channel in discord, where you can join us for faster support or just to chat, if you want to, of course._
